core/admin/src/main/scala/whisk/core/cli/Main.scala

```diff
@@ -0,0 +1,161 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package whisk.core.cli
+
+import java.io.File
+
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
+import ch.qos.logback.classic.{Level, LoggerContext}
+import org.rogach.scallop._
+import org.slf4j.LoggerFactory
+import pureconfig.error.ConfigReaderException
+import whisk.common.{AkkaLogging, Logging, TransactionId}
+import whisk.core.database.UserCommand
+
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.{Await, Future}
+import scala.util.{Failure, Success, Try}
+
+class Conf(arguments: Seq[String]) extends ScallopConf(arguments) {
+  banner("OpenWhisk admin command line tool")
+
+  val verbose = tally()
+  val configFile = opt[File](descr = "application.conf path")
```

Comment:
Hi @chetanmeh,

I'm wondering if there can be a _default_ path of `application.conf` that we can automatically generate this during ansible deployment? i.e. step on setup couchdb.

Reply:
That can be done. Another way I was thinking to have better support for Couch by read `whisk.properties` as being done by current `wskadmin`. So scalal tooling would also read that and seed in required typesafe properties such that for couch you need not pass in application.conf.

For others like Cosmos or later Mongo user can pass application.conf

Reply:
Reading `whisk.properties` also works fine, but is there any specific reason that we can't provide a consistent way? (not only supporting couch)
i.e. once user deploy/re-deploy/change the database, update the config for them.

Reply:
No specific reason. So. far I was thinking that we "publish" the tool jar to maven/github release and then admins can just download the jar and need not checkout whole of openwhisk repo to run the admin commands. In such a case they would need to provide a custom conf

Thinking about it it now it would be fine to support that. This may require following changes

1. Capture `ArtifactStore` settings as part of group vars
2. Have a `application.conf.j2` with some templating logic to render different config based on store type
3. Have it copied to std place say `${OPENWHISK_HOME}/whisk-cli.conf` or put it under `${OPENWHISK_HOME}/bin/whisk-cli.conf`
4. Have the tooling look there by default

Reply:
Cool, thanks! Just some comments, I think these can be done later. Let me know if you need more hands on these.

Reply:
Makes sense. I would try to get a base setup working with some parity with current wskadmin and get that merged. Then we can collaborate on such enhancements

Reply:
Good discussion. Had similar thoughts to @tz70s and look forward to future enhancements to make it easier to use.